hudi-spark-datasource/hudi-spark/src/main/antlr4/org/apache/hudi/spark/sql/parser/HoodieSqlCommon.g4

```diff
@@ -0,0 +1,87 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+grammar HoodieSqlCommon;
+
+ singleStatement
+    : statement EOF
+    ;
+
+statement
+    : compactionStatement                                              #compactionCommand
+    | .*?                                                              #passThrough
+    ;
+
+ compactionStatement
+    : operation = (RUN | SCHEDULE) COMPACTION  ON tableIdentifier (AT timestamp = NUMBER)?    #compactionOnTable
+    | operation = (RUN | SCHEDULE) COMPACTION  ON path = STRING   (AT timestamp = NUMBER)?    #compactionOnPath
```

Comment:
Was doing some research on paths we can take here. 
- Instead of extending SQL. Is it better to just surface this via stored procedures. `EXEC` statements? We can accept parameters easily? 
- I do see folks like snowflake using `ALTER table`  for clustering, because clustering keys are setup during table creation. But we probably want better flexibility. 
https://docs.snowflake.com/en/user-guide/tables-auto-reclustering.html#suspending-automatic-clustering-for-a-table

Reply:
In this model, how does the user configure compaction strategy and its input configs?

Reply:
I am afraid that spark sql do not support the EXEC statements. And also the `alter table` have different semantics with compaction and clustering.